articles/client-platforms/react/05-linking-accounts.md

```diff
@@ -0,0 +1,372 @@
+    this.auth = auth
+    // the Auth0Lock instance to show signin window to link a provider
+    this.lock = new Auth0Lock(auth.clientId, auth.domain, {
+      auth: {params: {state: 'linking'}}, // state to identify in the callback
+      allowedConnections: ['facebook', 'google-oauth2'],
+      languageDictionary: { // allows to override dictionary entries
+        title: 'Link with:' // new window title
+      }
+    })
+    this.link = this.link.bind(this)
+  }
+
+  link(){
+    // Call the show method to display the authentication window.
+    this.lock.show()
+  }
+}
+```
+
+`AuthService` continues responsible to listen to `authenticated` events, but we need a way to determine if the callback was trigged by a regular sign in process or by a linking. That's why in the code above you see the new `Auth0Lock` instance is receiving specific options, in that case `auth: {params: {state: 'linking'}}`. For more information about supported options please
+[check the documentation](https://github.com/auth0/lock/tree/v10.0.0-rc.1#authentication-options). Next you see the updated `AuthService` class where `_doAuthentication` callback is checking for `linking` status and calling `LinkAccount` method instead.
+
+```javascript
+/* ===== ./src/utils/AuthService.js ===== */
+import { EventEmitter } from 'events'
+import Auth0Lock from 'auth0-lock'
+
+export default class AuthService extends EventEmitter {
+  constructor(clientId, domain) {
+    super()
+    this.clientId = clientId
+    this.domain = domain
+    // Configure Auth0
+    this.lock = new Auth0Lock(clientId, domain)
+    // Add callback for lock `authenticated` event
+    this.lock.on('authenticated', this._doAuthentication.bind(this))
+    // Add callback for lock `authorization_error` event
+    this.lock.on('authorization_error', this._authorizationError.bind(this))
+    // binds login functions to keep this context
+    this.login = this.login.bind(this)
+  }
+
+  _doAuthentication(authResult){
+    authResult.state = authResult.state || '' //making sure state exists
+    if (authResult.state.includes('linking')){
+      this.linkAccount(authResult.idToken) // linkAccount when state is linking
+    } else {
+      // Otherwise saves the user token
+      this.setToken(authResult.idToken)
+      // Async loads the user profile data
+      this.lock.getProfile(authResult.idToken, (error, profile) => {
+        if (error) {
+          console.log('Error loading the Profile', error)
+        } else {
+          this.setProfile(profile)
+        }
+      })
+    }
+  }
+
+  ... // omitting methods
+
+  fetchApi(url, options){
+    // performs api calls sending the required authentication headers
+    const headers = {
+      'Accept': 'application/json',
+      'Content-Type': 'application/json',
+      'Authorization': 'Bearer ' + this.getToken()
+    }
+
+    const userId = this.getProfile().user_id
+    return fetch(`https://${this.domain}/api/v2/users/${userId}/${url}`, {
```

Comment:
The page fails to load: https://auth0-docs-content-pr-1721.herokuapp.com/docs/quickstart/spa/react/05-linking-accounts

Try escaping the `${some_text}` with `<%= "${some_text}" %>` -> https://github.com/auth0/docs#escaping-strings

Check the rest of the doc as well, there are more instances of `${}`